internal-enrichment/import-external-reference/src/import-external-reference.py

```diff
@@ -44,244 +64,830 @@ def __init__(self):
+
+        base = os.path.basename(path)
+        if not base:
+            base = parsed.netloc or "external-reference"
+
+        # Normalize to avoid Unicode compatibility issues (e.g., accents)
+        base = unicodedata.normalize("NFKD", base)
+        base = base.encode("ascii", "ignore").decode("ascii")
+
+        if self.timestamp_files:
+            ts = datetime.datetime.utcnow().strftime("_%Y%m%d_%H%M%S")
+            if suffix and base.endswith(suffix):
+                base = base[: -len(suffix)]
+            base += ts
+
+        # Ensure suffix is appended (but only once)
+        if suffix and not base.endswith(suffix):
+            base += suffix
+
+        # Truncate if too long (preserve suffix if present)
+        if len(base) > max_length:
+            if suffix and base.endswith(suffix):
+                base = base[: max_length - len(suffix)] + suffix
+            else:
+                base = base[:max_length]
+
+        return base
+
+    # ────────────────────────────────────────────────────────────────────────────────
+    # START / STOP
+    # ────────────────────────────────────────────────────────────────────────────────
+    async def _init_async(self):
+        """
+        Launch Playwright, spin up the browser, set up the queue and workers.
+        Does NOT start `listen()`.
+        """
+        self.helper.log_info("▶ entering _init_async")
+        self.loop = asyncio.get_running_loop()
+
+        # 1) start Playwright and the browser
+        pw = await async_playwright().start()
+        self.playwright = pw
+        self.helper.log_info("▶ launching browser…")
+        self.browser = await pw.chromium.launch(
+            headless=True,
+            args=[
+                "--disable-dev-shm-usage",
+                "--ignore-certificate-errors",
+                "--disable-blink-features=AutomationControlled",  # Disable WebDriver detection for WAF bypass
+            ],
+        )
+        self.helper.log_info("▶ browser launched")
+
+        # 2) task queue and workers
+        # Set task queue size to match worker count (prevents over-buffering)
+        self.task_queue = Queue(maxsize=self.worker_count)
+
+        self.helper.log_info("▶ browser workers starting")
+        self.workers = [
+            asyncio.create_task(self._browser_worker())
+            for _ in range(self.worker_count)
+        ]
+        self.helper.log_info(f"Started {self.worker_count} browser workers")
+
     def start(self):
-        self.helper.listen(message_callback=self._process_message)
+        # 1) new loop & set it
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+
+        # 2) bootstrap browser, queue, workers
+        loop.run_until_complete(self._init_async())
+
+        # 3) register clean-shutdown on Ctrl-C or SIGTERM
+        for sig in (signal.SIGINT, signal.SIGTERM):
+            loop.add_signal_handler(sig, lambda: loop.call_soon_threadsafe(loop.stop))
```

Comment:
@blauwers My previous comment didn't take into account this one.

As I said, testing on Windows, this part make the connector crash.

I suggest to handle this specific case for Windows OS based with

```suggestion
        for sig in (signal.SIGINT, signal.SIGTERM):
            try:
                loop.add_signal_handler(sig, lambda: loop.call_soon_threadsafe(loop.stop))
            except NotImplementedError:
                if sys.platform == "win32":
                    # Windows does not support add_signal_handler for most signals
                    pass
```

This workaround works and this is the results you see.

Here are some details on Python documentation:
- https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.add_signal_handler
- https://docs.python.org/3/library/intro.html#availability